From b1bb98164e4a61677f2efe10f89d3601d786d003 Mon Sep 17 00:00:00 2001
From: Reto Schneider <code@reto-schneider.ch>
Date: Sun, 30 Dec 2018 23:21:03 +0000
Subject: [PATCH] Add GARDENA smart Gateway mt7688 DTS option

---
 arch/mips/ralink/Kconfig | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/mips/ralink/Kconfig b/arch/mips/ralink/Kconfig
index 613d61763433..ba80e6e322cf 100644
--- a/arch/mips/ralink/Kconfig
+++ b/arch/mips/ralink/Kconfig
@@ -93,6 +93,11 @@ choice
 		depends on SOC_MT7620
 		select BUILTIN_DTB
 
+	config DTB_GARDENA_SMART_GATEWAY_MT7688
+		bool "Gardena smart Gateway MT7688"
+		depends on SOC_MT7620
+		select BUILTIN_DTB
+
 endchoice
 
 endif
-- 
2.11.0

