From ea1623d541b7820ee7c077fbeaba2af0827ddb51 Mon Sep 17 00:00:00 2001
From: Andrej Gessel <andrej.gessel@husqvarnagroup.com>
Date: Tue, 16 Oct 2018 17:29:04 +0200
Subject: [PATCH] add missing actions to Makefile

Signed-off-by: Andrej Gessel <andrej.gessel@husqvarnagroup.com>
---
 Makefile | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Makefile b/Makefile
index 490a38e..a6e368f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,13 @@
 EXTRA_CFLAGS += -Werror -DCONFIG_MT76_LEDS
 
+all:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD)
+
+clean:
+
+modules_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) modules_install
+
 obj-m := mt76.o
 obj-$(CONFIG_MT7603E) += mt7603/
 obj-$(CONFIG_MT76x02_LIB) += mt76x02-lib.o
