From 124a863431d2f910027f93d682246f177acfa48b Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Wed, 16 Oct 2019 22:44:05 +0200
Subject: [PATCH] MIPS: ralink: Add missing include

---
 arch/mips/ralink/mt7620.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/mips/ralink/mt7620.c b/arch/mips/ralink/mt7620.c
index c1ce6f43642b..952fea4d9d2e 100644
--- a/arch/mips/ralink/mt7620.c
+++ b/arch/mips/ralink/mt7620.c
@@ -19,6 +19,8 @@
 #include <asm/mach-ralink/mt7620.h>
 #include <asm/mach-ralink/pinmux.h>
 
+#include <asm/div64.h>
+
 #include "common.h"
 
 /* analog */
-- 
2.20.1

