From 1f66e4d822baded4a6625979ce1d73fdd504c2cd Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf@emagii.com>
Date: Mon, 28 Aug 2017 16:06:36 +0200
Subject: [PATCH] pci-mt7620.c: Use KERN_WARNING

Signed-off-by: Ulf Samuelsson <ulf@emagii.com>
---
 arch/mips/pci/pci-mt7620.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/mips/pci/pci-mt7620.c b/arch/mips/pci/pci-mt7620.c
index 628c513..d2540ab 100644
--- a/arch/mips/pci/pci-mt7620.c
+++ b/arch/mips/pci/pci-mt7620.c
@@ -33,7 +33,6 @@
 #define RALINK_GPIOMODE			0x60
 
 #define PPLL_CFG1			0x9c
-#define PDRV_SW_SET			BIT(23)
 
 #define PPLL_DRV			0xa0
 #define PDRV_SW_SET			(1<<31)
@@ -121,7 +120,7 @@ static int wait_pciephy_busy(void)
 		else
 			break;
 		if (retry++ > WAITRETRY_MAX) {
-			printk(KERN_WARN "PCIE-PHY retry failed.\n");
+			printk(KERN_WARNING "PCIE-PHY retry failed.\n");
 			return -1;
 		}
 	}
-- 
1.9.1

