From 8048b209a90cd88422cef435ea805f181032ebc6 Mon Sep 17 00:00:00 2001
From: Reto Schneider <code@reto-schneider.ch>
Date: Sun, 30 Dec 2018 22:39:27 +0000
Subject: [PATCH] MIPS: ralink: Enable SYS_SUPPORTS_ZBOOT

---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 9b13f31645b3..97138a76aa70 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -624,6 +624,7 @@ config RALINK
 	select RESET_CONTROLLER
 	select PINCTRL
 	select PINCTRL_RT2880
+	select SYS_SUPPORTS_ZBOOT
 
 config SGI_IP22
 	bool "SGI IP22 (Indy/Indigo2)"
-- 
2.11.0

