From 0b61fff92aeb4165aa741f29251aca8a6c69eee2 Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Tue, 26 Jun 2018 15:34:49 +0200
Subject: [PATCH] WIP: mips/arch: Add ethernet support

Signed-off-by: Stefan Roese <sr@denx.de>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index b32971c051..585e3a8397 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -73,6 +73,7 @@ config ARCH_MT7620
 	imply CMD_DM
 	select DISPLAY_CPUINFO
 	select DM
+	select DM_ETH
 	select DM_SERIAL
 	imply DM_SPI
 	imply DM_SPI_FLASH
