From a66d632566bafb167ae8af0798509f4aeb7ecbc4 Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Mon, 3 Sep 2018 18:01:37 +0200
Subject: [PATCH] WIP: MIPS: MT7620: Enable DM_GPIO Support

Signed-off-by: Stefan Roese <sr@denx.de>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 585e3a8397..3f667f69ef 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -74,6 +74,7 @@ config ARCH_MT7620
 	select DISPLAY_CPUINFO
 	select DM
 	select DM_ETH
+	imply DM_GPIO
 	select DM_SERIAL
 	imply DM_SPI
 	imply DM_SPI_FLASH
